app/src/main/java/org/oppia/android/app/topicdownloaded/TopicDownloadedViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package org.oppia.android.app.topicdownloaded
+
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.ObservableField
+import androidx.lifecycle.ViewModel
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.model.PlatformParameter
+import org.oppia.android.app.shim.IntentFactoryShim
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.platformparameter.MY_DOWNLOADS_FLAG
+import org.oppia.android.util.platformparameter.MY_DOWNLOADS_IS_DISABLE
+import org.oppia.android.util.platformparameter.PlatformParameterSingleton
+import javax.inject.Inject
+
+/** [ViewModel] for showing topic info details. */
+@FragmentScope
+class TopicDownloadedViewModel @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val platformParameterSingleton: PlatformParameterSingleton,
+  private val intentFactoryShim: IntentFactoryShim
+) : ObservableViewModel() {
+
+  /** Name of the downloaded topic */
+  val topicName = ObservableField<String>("")
+
+  /** Id of the profile which downloaded a topic */
+  var internalProfileId: Int = -1
+
+  /** Id of the downloaded topic */
+  lateinit var topicId: String
+
+  /** Starts TopicActivity with downloaded topic */
+  fun viewDownloadedTopic(@Suppress("UNUSED_PARAMETER") v: View) {
+    val disableMyDownloadsFeature = PlatformParameter.newBuilder()
+      .setName(MY_DOWNLOADS_FLAG)
+      .setBoolean(MY_DOWNLOADS_IS_DISABLE)
+      .build()
+
+    val platformParameterMap = mutableMapOf<String, PlatformParameter>()
+    platformParameterMap.put(disableMyDownloadsFeature.name, disableMyDownloadsFeature)
+    platformParameterSingleton.setPlatformParameterMap(platformParameterMap)
```

Comment:
This we **definitely** do not want to ever do. Changing state of platform parameters mid-app could cause really bad behaviors since the system is meant to guarantee immutability. We should be using a different code design to manage the state instead of changing the gate flag like this.

Reply:
fixed